src/BaGetter.Core/Configuration/MirrorOptions.cs

```diff
@@ -28,13 +28,65 @@ public class MirrorOptions : IValidatableObject
     [Range(0, int.MaxValue)]
     public int PackageDownloadTimeoutSeconds { get; set; } = 600;
 
+    public MirrorAuthenticationOptions Authentication { get; set; }
+
     public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
     {
-        if (Enabled && PackageSource == null)
+        if (!Enabled)
+        {
+            yield break;
+        }
+
+        if (PackageSource == null)
         {
             yield return new ValidationResult(
                 $"The {nameof(PackageSource)} configuration is required if mirroring is enabled",
-                new[] { nameof(PackageSource) });
+                [nameof(PackageSource)]);
+        }
+
+        if (Authentication != null)
+        {
+            if (Legacy && Authentication.Type is not (MirrorAuthenticationType.None or MirrorAuthenticationType.Basic))
+            {
+                yield return new ValidationResult(
+                    "Legacy v2 feeds only support basic authentication",
+                    [nameof(Legacy), nameof(Authentication)]);
+            }
+
+            switch (Authentication.Type)
+            {
+                case MirrorAuthenticationType.Basic:
+                    if (string.IsNullOrEmpty(Authentication.Username))
+                    {
+                        yield return new ValidationResult(
+                            $"The {nameof(Authentication.Username)} configuration is required for basic authentication",
+                            [nameof(Authentication.Username)]);
+                    }
+
+                    if (string.IsNullOrEmpty(Authentication.Password))
+                    {
+                        yield return new ValidationResult(
+                            $"The {nameof(Authentication.Password)} configuration is required for basic authentication",
+                            [nameof(Authentication.Password)]);
+                    }
+                    break;
+                case MirrorAuthenticationType.Bearer:
+                    if (string.IsNullOrEmpty(Authentication.Token))
+                    {
+                        yield return new ValidationResult(
+                            $"The {nameof(Authentication.Token)} configuration is required for bearer authentication",
+                            [nameof(Authentication.Token)]);
+                    }
+                    break;
+                case MirrorAuthenticationType.Custom:
+                    if (Authentication.CustomHeaders?.Count == 0)
```

Comment:
If custom is chosen but no `CustomHeaders` object is configured at all, this object would be null, right? In that case, this would return `false` as null != 0, not showing the error message. Please either add a null check or otherwise make sure a null also triggers the validation to fail.

Reply:
Good chatch. I've split it in two checks now.
Though now I'm not sure if the `.Count == 0` is even neccesary at all.
In most cases if Custom Auth is set the user probably also wants to add headers, but leaving the object empty technically is not an invalid configuration.
Basically it should only be warning, but the validation has afaik no option for warnings.